kernel/sched.c

```diff
@@ -541,6 +541,25 @@ static void ready_thread(struct k_thread *thread)
 	}
 }
 
+#ifdef CONFIG_EVENTS
+void z_ready_linked_threads(struct k_thread *thread)
+{
+	/*
+	 * Ready all the threads linked together through the
+	 * [next_link] field.
+	 */
+
+	LOCKED(&sched_spinlock) {
```

Comment:
Oh, but to be clear, having just realized this: you do need to have that loop in a lock of some form (e.g. under the event's own lock). You can't let the readied threads preempt the current thread, because as the code currently works (using a temporary linked list) the threads are readied in the "wrong" order, from lowest priority to highest.